### PR TITLE
Amp ad element

### PR DESCRIPTION
**Pull request by nicl**
## What does this change?

Populate ad placeholders with actual ads! The was started by @philmcmahon with a bit of finishing by me.

I'm sure there are bugs here so I'm going to chat with commercial to get more feedback, but I think it will probably be easiest to discover most issues on prod.

## Why?

Crucial for go-live.

screenshot 2019-02-11 at 11 48 17

### CODE REVIEW

File: packages/frontend/amp/components/elements/AdComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+            sans-serif;
+        padding: 3px 10px;
+        color: ${palette.neutral[46]};
+        text-align: right;
+    }
+`;
+
+const dfpAdUnitRoot = 'theguardian.com';
+
+// type AdRegion =  'US' | 'ROW' | 'AU';
```
Comment (philmcmahon) [inline packages/frontend/amp/components/elements/AdComponent.tsx]: remove?

File: packages/frontend/amp/components/elements/AdComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+// const editionToAdRegion(edition: Edition): AdRegion
+
+const ampData = (
+    section: string,
+    dfpAccountId: string,
+    contentType: string,
+): string => {
+    const dfpAccountId2 = '59666047';
```
Comment (philmcmahon) [inline packages/frontend/amp/components/elements/AdComponent.tsx]: this should be properly passed through via some kind of config thign

File: packages/frontend/amp/components/elements/AdComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    switches,
+    commercialProperties,
+}) => (
+    <div className={adStyle}>
+        <amp-ad
+            width={300}
+            height={250}
+            data-npa-on-unknown-consent={true}
+            data-loading-strategy={'prefer-viewability-over-views'}
+            // data-slot={AmpAdDataSlot(article).toString()}
```
Comment (philmcmahon) [inline packages/frontend/amp/components/elements/AdComponent.tsx]: remove

File: packages/frontend/amp/components/elements/AdComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+            layout={'responsive'}
+            type={'doubleclick'}
+            json={adJson(edition, commercialProperties.editionAdTargeting)}
+            data-slot={ampData(section, dfpAccountId, contentType)}
+            rtc-config={realTimeConfig('test', edition, false, switches)}
```
Comment (philmcmahon) [inline packages/frontend/amp/components/elements/AdComponent.tsx]: 'test' is not the correct prebidserverurl - will need to come from config :(

File: packages/frontend/amp/lib/ad-json.test.ts [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        };
+        const res = adJson(edition, [targetings]);
+        const p = res.targeting.find(param => param.name === 'p');
+        if (p === undefined) {
+            return fail();
+        }
+
+        expect(p.value).toBe('amp');
+    });
+});
```
Comment (philmcmahon) [inline packages/frontend/amp/lib/ad-json.test.ts]: tests! What a hero!
 hero

**Comment by nicl** [comment]
@philmcmahon hey thanks. I've inlined most of the config now with the plan to do config properly (for this and a load of other stuff) after go live. Does that sound okay?

File: packages/frontend/amp/components/elements/AdComponent.tsx [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+            data-loading-strategy={'prefer-viewability-over-views'}
+            layout={'responsive'}
+            type={'doubleclick'}
+            json={adJson(edition, commercialProperties.editionAdTargeting)}
+            data-slot={ampData(section, contentType)}
+            rtc-config={realTimeConfig(
+                edition,
+                config.useKrux,
+                config.usePrebid,
+            )}
```
Comment (SiAdcock) [inline packages/frontend/amp/components/elements/AdComponent.tsx]: Is there any metadata that distinguishes ads appearing on dotcom-rendering from those appearing on frontend?

**Pull request merged**